libs/vscode/project-details/src/lib/new-project-details-preview.ts

```diff
@@ -0,0 +1,560 @@
+import { PDVData } from '@nx-console/shared/types';
+import { onWorkspaceRefreshed } from '@nx-console/vscode/lsp-client';
+import { getPDVData } from '@nx-console/vscode/nx-workspace';
+import { join } from 'path';
+import {
+  commands,
+  Disposable,
+  ExtensionContext,
+  Uri,
+  ViewColumn,
+  WebviewPanel,
+  window,
+} from 'vscode';
+import {
+  assign,
+  createActor,
+  enqueueActions,
+  fromPromise,
+  setup,
+} from 'xstate';
+import { ProjectDetailsPreview } from './project-details-preview';
+import { handleGraphInteractionEventBase } from '@nx-console/vscode/graph-base';
+import { getGraphWebviewManager } from '@nx-console/vscode/project-graph';
+
+export class NewProjectDetailsPreview implements ProjectDetailsPreview {
+  private webviewPanel: WebviewPanel;
+
+  projectRoot: string | undefined;
+
+  constructor(private path: string, private context: ExtensionContext) {
+    this.webviewPanel = window.createWebviewPanel(
+      'nx-console-project-details',
+      `Project Details`,
+      ViewColumn.Beside,
+      {
+        enableScripts: true,
+      }
+    );
+    const machine = setup({
```

Comment:
Not really an XState expert, but it feels like the render side-effects aren't part of the workflow. Maybe instead it should be a handler to `actor.subscribe` that renders the webview depending on the current state?